packages/react-native/android/src/main/java/com/bugsnag/reactnative/BugsnagReactNative.java

```diff
@@ -1,103 +1,193 @@
+    static String bugsnagAndroidVersion;
 
-    public BugsnagReactNative(ReactApplicationContext reactContext) {
+    private final BreadcrumbDeserializer breadcrumbDeserializer = new BreadcrumbDeserializer();
+
+    public BugsnagReactNative(@Nonnull ReactApplicationContext reactContext) {
         super(reactContext);
-        this.reactContext = reactContext;
-    }
-
-    // @Override
-    // public Map<String, Object> getConstants() {
-    //   HashMap<String, Object> constants = new HashMap<>();
-    //   Client client = getClient();
-    //   if (client != null) {
-    //     Configuration config = client.getConfig();
-    //     String key = config.getApiKey();
-    //     constants.put("apiKey", key);
-    //   }
-    //   // TODO: if bugsnag has not been started, it would be nice to read the manifest
-    //   // and get the default api key. This is how it works on iOS.
-    //   return constants;
-    // }
-    //
-    // // @Override
-    // public boolean hasConstants() {
-    //   return true;
-    // }
-    //
+    }
+
+    /**
+     * Instantiates a bugsnag client using the API key in the AndroidManifest.xml
+     *
+     * @param context the application context
+     * @return the bugsnag client
+     */
+    public static Client start(Context context) {
+        Client client = Bugsnag.init(context);
+        configureClient(client);
+        return client;
+    }
+
+    /**
+     * Instantiates a bugsnag client with a given API key.
+     *
+     * @param context the application context
+     * @param apiKey  the api key for your project
+     * @return the bugsnag client
+     */
+    public static Client startWithApiKey(Context context, String apiKey) {
+        Client client = Bugsnag.init(context, apiKey);
+        configureClient(client);
+        return client;
+    }
+
+    /**
+     * Instantiates a bugsnag client with a given configuration object.
+     *
+     * @param context the application context
+     * @param config  configuration for how bugsnag should behave
+     * @return the bugsnag client
+     */
+    public static Client startWithConfiguration(Context context, Configuration config) {
+        Client client = Bugsnag.init(context, config);
+        configureClient(client);
+        return client;
+    }
+
+    private static void configureClient(Client client) {
+        InternalHooks.configureClient(client);
+
+        // TODO set these as config up front if possible
+        client.setIgnoreClasses("com.facebook.react.common.JavascriptException");
+        bugsnagAndroidVersion = client.getClass().getPackage().getSpecificationVersion();
+        // TODO set codeBundleId here
+        Log.d("BugsnagReactNative", "Initialised bugsnag-react-native");
+    }
+
     @Override
     public String getName() {
         return "BugsnagReactNative";
     }
 
-    /** Start a new session. */
+    @ReactMethod(isBlockingSynchronousMethod = true)
+    public void getConfig(Promise promise) {
```

Comment:
One more thing, this method signature should be `public WritableMap getConfig()`. Then this should be good to go.

Reply:
Updated the method signature as requested.